File: TUW_Radar_Image_Compositing/R/gee_uc1_temp.R
library(openeo)
library(tibble)

gee = connect(host = backend_url,user = username,password = password,login_type = "basic")

collections = list_collections()

s1 = collections$`COPERNICUS/S1_GRD`
dims = dimensions(s1)
processes = list_processes()
p = processes()

march = p$load_collection(id = s1,
                          spatial_extent = list(west=16.06,south=48.1,east=16.65,north=48.31),
                          temporal_extent = c("2017-03-01","2017-04-01"),
                          bands = c("VV"))

april = p$load_collection(id = s1,
                          spatial_extent = list(west=16.06,south=48.1,east=16.65,north=48.31),
                          temporal_extent = c("2017-04-01","2017-05-01"),
                          bands = c("VV"))
may = p$load_collection(id = s1,
                          spatial_extent = list(west=16.06,south=48.1,east=16.65,north=48.31),
                          temporal_extent = c("2017-05-01","2017-06-01"),
                          bands = c("VV"))

mean_march = p$reduce_dimension(data = march,dimension = dims$t,reducer = function(x, context) {
  mean(x)})
mean_april = p$reduce_dimension(data = april,dimension = dims$t,reducer = function(x, context) {
  mean(x)})
mean_may = p$reduce_dimension(data = may,dimension = dims$t,reducer = function(x, context) {
  mean(x)})

R_band = p$rename_labels(data = mean_march, dimension = "bands", target = c("R"), source = array())
G_band = p$rename_labels(data = mean_april, dimension = "bands", target = c("G"), source = array())
B_band = p$rename_labels(data = mean_may, dimension = "bands", target = c("B"), source = array())

RG = p$merge_cubes(cube1 = R_band, cube2 = G_band)
RGB = p$merge_cubes(cube1 = RG, cube2 = B_band)

final = p$save_result(data = RGB,format = "GTIFF-THUMB")

compute_result(graph=graph,output_file = "uc1_temp_comp_gee.tif")

File: TUW_Radar_Image_Compositing/Python/gee_uc1_temp.py
import openeo

GEE_DRIVER_URL = "https://earthengine.openeo.org/v1.0"

user = "group2"
password = "test123"

# Connect to backend via basic authentication
con = openeo.connect(GEE_DRIVER_URL)
con.authenticate_basic(user, password)

datacube = con.load_collection("COPERNICUS/S1_GRD",
                               spatial_extent={"west": 16.06, "south": 48.10, "east": 16.65, "north": 48.31, "crs": 4326},
                               temporal_extent=["2017-03-01", "2017-06-01"],
                               bands=["VV"])
march = datacube.filter_temporal("2017-03-01", "2017-04-01")
april = datacube.filter_temporal("2017-04-01", "2017-05-01")
may = datacube.filter_temporal("2017-05-01", "2017-06-01")

mean_march = march.mean_time()
mean_april = april.mean_time()
mean_may = may.mean_time()

R_band = mean_march.rename_labels(dimension="bands", target=["R"])
G_band = mean_april.rename_labels(dimension="bands", target=["G"])
B_band = mean_may.rename_labels(dimension="bands", target=["B"])

RG = R_band.merge(G_band)
RGB = RG.merge(B_band)

datacube = RGB.save_result(format="GTIFF-THUMB")

job = datacube.send_job()
job.start_and_wait().download_results()
